### ISSUE TITLE
git lfs migrate import --no-rewrite claims my files have no matching .gitattributes pattern

### ISSUE THREAD

**Issue by l-abels** [git-lfs/git-lfs#4054]
**Describe the issue**
I'm trying to migrate the image files in my repo to git lfs. The folder structure looks like this:
```
/
 ui/
  __image_snapshots__/
    image.png
  .gitattributes   
```

The `.gitattributes` file was created by `git lfs track "*.png"` and contains only one line, `*.png filter=lfs diff=lfs merge=lfs -text`

When I run `git lfs migrate import --no-rewrite ./__image_snapshots__/image.png` I get the following error:
```
fatal: file ./__image_snapshots__/image.png did not match any Git LFS filters in .gitattributes
```

**System environment**
OS: MacOS Mojave

**Comment by bk2204** [comment]
Hey,

It looks like you have a `ui` directory above your `__image_snapshots__` directory.  Does it work if you use `git lfs migrate import --no-rewrite ui/__image_snapshots__/image.png`?

**Comment by l-abels** [comment]
I get the same error.  It might be worth noting that I'm running these commands in the `ui` directory.

**Comment by l-abels** [comment]
I turned on `GIT_TRACE` and captured this relevant portion:
```
14:42:55.124779 trace git-lfs: filepathfilter: rewrite "ui/*.png" as "ui/*.png"
14:42:55.125576 trace git-lfs: filepathfilter: rejecting "./ui/__image_snapshots__/image.png" via [ui/*.png]
```

**Comment by l-abels** [comment]
I resolved the issue by moving my `.gitattributes` file to the top level of my repository.

**Comment by yegorius** [comment]
Having the same problem. My `.gitattributes` is in the root directory of the repo though.

**Comment by JacquesOfAllTrades** [comment]
Same problem here, using Git version `2.30.2.windows.1`, and LFS version `git-lfs/2.13.2 (GitHub; windows amd64; go 1.14.13; git fc664697)`.

Attempted migration using a command like (paraphrased for privacy and brevity):
`git lfs migrate import --no-rewrite -m "message" foo/bar/file.css.gz`
and it failed with this error:
`fatal: file foo/bar/file.css.gz did not match any Git LFS filters in .gitattributes`

However, I got it to work by changing the file specifier pattern to the same style as `.gitignore`. That is, it failed with this rule:
   `*.gz filter=lfs diff=lfs merge=lfs -text`
But it worked with this one: 
   `**/*.gz filter=lfs diff=lfs merge=lfs -text`

Neither the official documentation for `.gitattributes` nor any third-party guides I've seen indicate that this is even supported, much less required. (The docs are frustratingly vague about this point, though.)

Importantly, note that running `git lfs migrate import` **without** the `--no-rewrite` option works fine on the same repo, without adding `**/` to the patterns in `.gitattributes`. (Unfortunately I can't rewrite history on this particular repo, so I need to use `--no-rewrite`.)

I have to wonder if `lfs migrate import --no-rewrite` got changed to use the same pattern-matching logic as Git itself, because I'd assume that this must've worked at some point.

**Comment by bk2204** [comment]
This will likely be fixed in 3.0.0, where we're improving the pattern matching.

**Comment by timseries** [comment]
This does seem to have been fixed on git-lfs 3.0.2 (encountered the problem before upgrading from 2.13.3).

**Comment by bk2204** [comment]
Good.  I'll close this, then.
